src/main/java/org/chocosolver/solver/search/IResolutionHelper.java

```diff
@@ -0,0 +1,384 @@
+ * @since 25/04/2016.
+ */
+public interface IResolutionHelper extends ISelf<Model> {
+
+	/**
+	 * Attempts to find a solution of the declared problem.
+	 * <ul>
+	 * <li>If the method returns <tt>null</tt>:</li>
+	 * <ul>
+	 * <li>either a stop criterion (e.g., a time limit) stops the search before a solution has been found,</li>
+	 * <li>or no solution exists for the problem (i.e., over-constrained).</li>
+	 * </ul>
+	 * <li>if the method returns a {@link Solution}:</li>
+	 * <ul>
+	 * <li>a solution has been found. This method can be called anew to look for the next solution, if any.</li>
+	 * </ul>
+	 * </ul>
+	 * <p>
+	 * If a solution has been found, since the search process stops on that solution, variables' value can be read, e.g.,
+	 * {@code intvar.getValue()} or the solution can be recorded:
+	 *
+	 * <pre>
+	 * {
+	 * 	&#64;code
+	 * 	Solution s = new Solution(model);
+	 * 	s.record();
+	 * }
+	 * </pre>
+	 * <p>
+	 * This method run the following instructions:
+	 *
+	 * <pre>
+	 *     {@code
+	 *     if(_me().solve()) {
+	 *          return new Solution(_me()).record();
+	 *     }else{
+	 *          return null;
+	 *       }
+	 *     }
+	 * </pre>
+	 *
+	 * @param stop
+	 *          optional criterions to stop the search before finding all/best solution
+	 * @return a {@link Solution} if and only if a solution has been found, <tt>null</tt> otherwise.
+	 */
+	default Solution findSolution(Criterion... stop) {
+		if (stop != null) {
+			_me().getSolver().addStopCriterion(stop);
+		}
+		boolean found = _me().solve();
+		if (stop != null) {
+			_me().getSolver().removeStopCriterion(stop);
+		}
+		if (found) {
+			return new Solution(_me()).record();
+		}else{
+			return null;
+		}
+	}
+
+	/**
+	 * Attempts to find all solutions of the declared problem.
+	 * <ul>
+	 * <li>If the method returns an empty list:</li>
+	 * <ul>
+	 * <li>either a stop criterion (e.g., a time limit) stops the search before any solution has been found,</li>
+	 * <li>or no solution exists for the problem (i.e., over-constrained).</li>
+	 * </ul>
+	 * <li>if the method returns a list with at least one element in it:</li>
+	 * <ul>
+	 * <li>either the resolution stops eagerly du to a stop criterion before finding all solutions,</li>
+	 * <li>or all solutions have been found.</li>
+	 * </ul>
+	 * </ul>
+	 * <p>
+	 * This method run the following instructions:
+	 *
+	 * <pre>
+	 * {
+	 * 	&#64;code
+	 * 	List<Solution> solutions = new ArrayList<>();
+	 * 	while (model.solve()) {
+	 * 		solutions.add(new Solution(model).record());
+	 * 	}
+	 * 	return solutions;
+	 * }
+	 * </pre>
+	 *
+	 * @param stop
+	 *          optional criterions to stop the search before finding all/best solution
+	 * @return a list that contained the found solutions.
+	 */
+	default Stream<Solution> findAllSolutions(Criterion... stop) {
+		if (stop != null) {
+			_me().getSolver().addStopCriterion(stop);
+		}
+		Spliterator<Solution> it = new Spliterator<Solution>() {
```

Comment:
can we consider create an internal class (or external) for the SplitIterator, that would certainly be helpful for other methods, and it simplifies this code